### PR TITLE
Per file author distribution

### LINKED ISSUE #52: Use Person interface instead of string for author and committer
in [parser/src/model.ts](https://github.com/git-visual/git-visual/pull/47/files#diff-ef18a8dbbef15227f5b12f96b539c3ddce94c2dac93cd1f8b3c8ca4b5f5947a3)

### LINKED ISSUE #59: Per file author distribution

### LINKED ISSUE #68: Add log module for different log levels

### PR DESCRIPTION
**Pull request by tjomson**
resolves #52 
resolves #59 
resolves #68

### CODE REVIEW

**Review by tjomson — COMMENTED** [review]
Should path in data.ts be changed to 
> "../../parser/.temp/git-visual_main.json"

to match where the parser outputs it by default?
Also, if i do that, i get an error that parent2 i missing on a type.

When i run the parser, and then the frontend, all circles are the same size, is this intended?
If not, it probably is related to that error

File: parser/src/authorship.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+const emptyGitTree = "4b825dc642cb6eb9a060e54bf8d69288fbee4904"
+
+export async function hydrateTreeWithAuthorship(
+  repo: string,
+  commit: GitCommitObject
+) {
+  let { hash: child, parent } = commit
+
+  // eslint-disable-next-line no-constant-condition
```
[the remark this answers is not in the document]
Comment (joglr) [reply parser/src/authorship.ts:14]: It is no longer relevant

[the remark this answers is not in the document]
Comment (hojelse) [reply parser/src/authorship.ts:14]: Removed

File: parser/src/authorship.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import { log } from "./log.js"
+import { GitCommitObject } from "./model.js"
+import { parseCommit, parseCommitLight } from "./parse.js"
+import { gitDiffNumStatParsed, lookupFileInTree } from "./util.js"
+
+const emptyGitTree = "4b825dc642cb6eb9a060e54bf8d69288fbee4904"
+
+export async function hydrateTreeWithAuthorship(
```
[the remark this answers is not in the document]
Comment (hojelse) [reply parser/src/authorship.ts:8]: Jep :)

**Review by hojelse — APPROVED** [review]
lgtm :rocket:

**Pull request merged**